src/catch2/internal/catch_test_registry.hpp

```diff
@@ -47,6 +47,33 @@ Detail::unique_ptr<ITestInvoker> makeTestInvoker( void (C::*testAsMethod)() ) {
     return Detail::make_unique<TestInvokerAsMethod<C>>( testAsMethod );
 }
 
+template <typename C>
+class TestInvokerFixture : public ITestInvoker {
+    void ( C::*m_testAsMethod )();
+    Detail::unique_ptr<C> m_fixture = nullptr;
+
+public:
+    TestInvokerFixture( void ( C::*testAsMethod )() ) noexcept : m_testAsMethod( testAsMethod ) {}
+
+    void testCaseStarting() override { 
+        m_fixture = Detail::make_unique<C>(); 
+    }
+
+    void testCaseEnding() override { 
+        m_fixture.reset(); 
+    }
+
+    void invoke() const override {
+        auto* f = const_cast<C*>( m_fixture.get() );
```

Comment:
Just want to draw attention to this because const_cast is usually a sin and UB. However, this is a valid use-case of it that does not invoke UB since m_fixture does not point to a const object. This const_cast is required due to how Detail::unique_ptr implements get(). It differs from std::unique_ptr since it will return a pointer to const if the unique_ptr itself is const. This is not what std::unique_ptr does. https://en.cppreference.com/w/cpp/memory/unique_ptr/get.

The other option was to manually manage a raw pointer and add special member functions to ensure that it is handled correctly, but I have erred with const_cast in this case. Happy to change this though

Reply:
The implementation difference is intentional, the shallow const on language pointers makes it annoyingly easy to throw away logical constness. So the const cast here is fine-ish.

Here is a thought though: why not make the fixture member function const?

It would be inconsistent with the `TEST_CASE_METHOD` macro, but this is fine; inconsistency can be okay if it brings improvement. The advantage is that it would be much harder to accidentally mutate the fixture -> this is much more important here than for `TEST_CASE_METHOD`, as the fixture here is persistent between test runs. It is still possible to allow fixture mutation, by marking the intended-to-change member variable with `mutable`.

I actually took your branch and tried a quick sketch of this and I think it is better, but I am willing to hear counter arguments.

Reply:
I have zero issues with this honestly. I agree that this would be a better course of action and would make test writers  think about what they want to be mutatable. Const as the default is never usually a bad thing. I will make this change, and also note it in the docs, and perhaps also make a example to demonstrate it. Thanks for the great improvement!